### PR TITLE
allow ASCIIDOCTOR_CORE_VERSION to be ref in any fork

### PR DESCRIPTION
**Pull request by mojavelinux**

### CODE REVIEW

File: npm/builder.js [hunk 1]
```diff
@@ -27,10 +27,16 @@ function Builder () {
     'node_modules/opal-runtime/src/opal.js',
     'build/asciidoctor-lib.js'
   ];
-  if (process.env.ASCIIDOCTOR_CORE_VERSION) {
-    this.asciidoctorCoreVersion = process.env.ASCIIDOCTOR_CORE_VERSION;
+  var asciidoctorRef = process.env.ASCIIDOCTOR_CORE_VERSION;
+  if (asciidoctorRef && ~asciidoctorRef.indexOf('/')) {
```
Comment (Mogztter) [inline npm/builder.js]: In my opinion, the tilde operator is uncommon, could you use `includes` instead ? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes

File: npm/builder.js [hunk 2]
```diff
@@ -27,10 +27,16 @@ function Builder () {
     'node_modules/opal-runtime/src/opal.js',
     'build/asciidoctor-lib.js'
   ];
-  if (process.env.ASCIIDOCTOR_CORE_VERSION) {
-    this.asciidoctorCoreVersion = process.env.ASCIIDOCTOR_CORE_VERSION;
+  var asciidoctorRef = process.env.ASCIIDOCTOR_CORE_VERSION;
```
Comment (Mogztter) [inline npm/builder.js]: You can use `const` and `let` because we are using Node 6+

Comment (Mogztter) [inline npm/builder.js]: I think we should add an example in a comment to mention that `ASCIIDOCTOR_CORE_VERSION` can be a reference to a fork (user/project#branch) or a tag (v1.5.6.1).

Comment (Mogztter) [inline npm/builder.js]: You can use `const` and `let` because we are using Node 6+

File: npm/builder.js (hunk 1 above)
Comment (Mogztter) [inline npm/builder.js]: In my opinion, the tilde operator is uncommon, could you use `includes` instead ? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes

File: npm/builder.js (hunk 2 above)
Comment (Mogztter) [inline npm/builder.js]: I think we should add an example in a comment to mention that `ASCIIDOCTOR_CORE_VERSION` can be a reference to a fork (user/project#branch) or a tag (v1.5.6.1).

File: npm/builder.js [hunk 3]
```diff
@@ -27,10 +27,16 @@ function Builder () {
...
+  if (asciidoctorRef && ~asciidoctorRef.indexOf('/')) {
+    var segments = asciidoctorRef.split('/');
+    this.asciidoctorCoreUser = segments[0];
+    segments = segments[1].split('#');
+    this.asciidoctorCoreRepo = segments[0];
```
Comment (Mogztter) [inline npm/builder.js]: `asciidoctorCoreRepo` should always be `asciidoctor` right ?

File: npm/builder.js (hunk 1 above)
Comment (mojavelinux) [reply npm/builder.js]: Yep.

File: npm/builder.js (hunk 2 above)
Comment (mojavelinux) [reply npm/builder.js]: :+1:

File: npm/builder.js (hunk 3 above)
Comment (mojavelinux) [reply npm/builder.js]: If we're going to allow the form user/repo#branch, we should do it properly and not make any assumptions.

**Comment by mojavelinux** [comment]
Updated!

**Pull request merged**